qcengine/util.py

```diff
@@ -457,9 +457,12 @@ def which_import(plug, return_bool=False):
 
     """
     import pkgutil
-    plug_spec = pkgutil.find_loader(plug)
+    module_spec = pkgutil.find_loader(module)
+
+    if module_spec is None:
+        if raise_error:
+            raise ModuleNotFoundError(f"Could not find module '{module}'' in Python path.")
```

Comment:
fwiw, I tried and rejected the 3-mode return from `which_import` of bool, path/None, raise_error in favor of handling only the first two at this level (remember `which` and `which_import` need to follow the same pattern). Most of the time, you want to raise a more specific error and so you'd end up catching and re-raising anyways. just my opinion.

Reply:
Hmm, the 99% case seems to be that you want to throw here? Easy to do in `which` as well. 

The Psi4 logic that you linked to above is reproduced in these lines. I didn't think rewriting the same thing over and over was healthy.

Reply:
Yes, often you want to throw, but you want a program specific message like https://github.com/MolSSI/QCElemental/blob/master/qcelemental/models/molecule.py#L236 (as opposed to the rather bad one I linked to for psi).

For the fallback case when the generic error will do, I'm good with the `raise_error` arg in `which`/`which_import` like you have. I'll implement that in the qcel version. But for most in-module calls to which*, I think we should be using the T/F and writing good error strings (e.g., `conda install psi4 -c psi4`. That was my aim in localizing all the checks to each engine's `.found` method.

Side question is should there be a generalized error? `which_import` would be `ModuleNotFoundError` but `which` is natively just `None`. And you don't want the error changing when go btwn `psi4` and `import psi4`. (These are some of the reasons I punted on raise_error before.)

Reply:
I say `which` should just raise a `ModuleNotFoundError` (IMO). Functionally the same thing here.

Reply:
Ok, that will require no changes to https://github.com/MolSSI/QCElemental/pull/63 then.

How pressing is this PR? Easiest would be to wait on new qcel release, then #63 merge, then rebase, then patch up the `found` messages. But that's days away.

Reply:
Not a huge hurry at the moment.